### ISSUE TITLE
Implement a tag management user interface

### ISSUE THREAD

**Issue by thomasjo** [wandb/client#1439]
_NOTE: I'm opening this feature request here as requested by @lukas after a brief interaction on Twitter._

## Description

The tag feature currently implemented on [wandb.ai](https://wandb.ai) is very useful for keeping lots of experiment runs organized. I personally use this for tracking which datasets, partitioning schemes, and similar were used for individual runs. The type of information that doesn't really make sense to "shoehorn" into the config/hyperparameter tracking functionality. More importantly, using tags makes it extremely fast and convenient to filter and search for runs. I'm sure there are countless other useful ways to use tags, which is another strength of this feature.

The biggest drawback of the current tag functionality is that there is no user interface for managing tags. Currently, you can only create new tags. There is no way to delete or edit tags. Other missing functionality includes customization of tag color and the display order of tags in lists.

## Proposed feature design

I think the easiest way to initially approach the design and implementation of a tag management user interface would be to look at the _issue label_ user interface on GitHub.

Here's what a list of labels might look like on a typical repository on GitHub:
![labels-list](https://user-images.githubusercontent.com/3622/97568026-2d1dce80-19e7-11eb-817e-6f3572fb7074.png)

The interface makes it very fast and convenient to create new labels without leaving the list view:
![labels-new](https://user-images.githubusercontent.com/3622/97568744-5c344000-19e7-11eb-8010-0320db093cef.png)

Edit existing labels is also fast and convenient with an in-place approach:
![labels-edit](https://user-images.githubusercontent.com/3622/97569150-7837e180-19e7-11eb-89e5-26b920f85b4c.png)

Additionally from this list view, it's also easy to delete labels, and seeing the number of open issues or PRs associated with each label. The only thing missing is the ability to customize the display order. However, since GitHub displays labels in lexicographic order, the display order can be controlled by prefixing label names with digits or similar "hacks."

In summary, I think the current implementation of issue labels would be a reasonable design to take inspiration from when implementing a tag management interface on [wandb.ai](https://wandb.ai).

**Comment by cvphelps** [comment]
Hi Thomas, thanks for the thoughtful and well-documented feature request. This is a good idea for an enhancement. It sounds like the key requirements are:
- Delete a tag from the list, removing it from all related runs
- Rename a tag, updating it everywhere
- Change the color of a tag manually

**Comment by thomasjo** [comment]
@cvphelps Thanks for summarizing my feature request as a list of actionable items. Your summary is perfect.

**Comment by soheeyang** [comment]
Hi, any update on this? I'm looking for a way to rename and/or delete tags, but I can't find a way to do so.

**Comment by veronica320** [comment]
+1, any update on this?
